Index: include/log4cxx/helpers/tchar.h
===================================================================
RCS file: /home/cvspublic/logging-log4cxx/include/log4cxx/helpers/tchar.h,v
retrieving revision 1.35
diff -c -r1.35 tchar.h
*** include/log4cxx/helpers/tchar.h	13 May 2004 21:14:38 -0000	1.35
--- include/log4cxx/helpers/tchar.h	21 May 2004 22:56:44 -0000
***************
*** 42,64 ****
  		return dst;
  	}
  	
! 	static void int64ToString(wchar_t * dst, size_t maxlen, const int64_t& ll)
  	{
  #ifdef WIN32
! 		_snwprintf(dst, maxlen, L"%I64d", ll);
  #else
! 		swprintf(dst, maxlen, L"%lld", ll);
  #endif
  	}
! 	
! 	static void int64ToString(char * dst, size_t maxlen, const int64_t& ll)
  	{
  #ifdef WIN32
! 		_snprintf(dst, maxlen, "%I64d", ll);
  #else
! 		snprintf(dst, maxlen, "%lld", ll);
  #endif
  	}
  };
  
  namespace std 
--- 42,74 ----
  		return dst;
  	}
  	
!          static void int64ToString(char * dst, size_t maxlen, const int64_t& ll)
  	{
  #ifdef WIN32
! 		_snprintf(dst, maxlen, "%I64d", ll);
  #else
! 		snprintf(dst, maxlen, "%lld", ll);
  #endif
  	}
!     
! #ifdef __CYGWIN__
!          static void int64ToString(wchar_t * dst, size_t maxlen, const int64_t& ll)
! 	{
! 	  char temp[32];
! 	  int64ToString(temp, sizeof(temp), ll);
! 	  ansiToUnicode(dst, temp);
! 	}
! #else
!          static void int64ToString(wchar_t * dst, size_t maxlen, const int64_t& ll)
  	{
  #ifdef WIN32
! 		_snwprintf(dst, maxlen, L"%I64d", ll);
  #else
! 		swprintf(dst, maxlen, L"%lld", ll);
  #endif
  	}
+ #endif /* __CYGWIN__ */
+ 
  };
  
  namespace std 
